sql-parser/src/main/java/io/crate/sql/tree/FrameBound.java

```diff
@@ -75,17 +73,14 @@
             @Override
             public <T> int getStart(int pStart,
                                     int pEnd,
-                                    int currentFrameStartIdx,
                                     int currentRowIdx,
                                     @Nullable Comparator<T> cmp,
                                     List<T> rows) {
                 if (pStart == currentRowIdx) {
                     return pStart;
                 } else {
                     if (cmp != null) {
-                        return arePeers(rows, currentFrameStartIdx, currentRowIdx, cmp) ?
-                            currentFrameStartIdx :
-                            currentRowIdx;
+                        return findFirstPreviousPeer(rows, currentRowIdx, cmp);
```

Comment:
Could this potentially go past the partition boundary?

Reply:
I initially thought that's not possible but ordering on different criteria to partitioning _could_ make it happen. Guarded against it in the FrameBound to take the partition start if it happens https://github.com/crate/crate/pull/8759/files#diff-73b3f1b277b1006e2a16e8c351147febR83 and I added a test for this